vignettes/speedup_options.Rmd

```diff
@@ -0,0 +1,458 @@
+)
+
+# No forecasting; just estimation
+horizon <- 0
+```
+
+Now, let's generate the `estimates` object from `epinow()`.
+```{r estimates}
+estimates <- epinow(
+  data = example_confirmed[1:100],
+  generation_time = generation_time_opts(example_generation_time),
+  delays = delay_opts(example_incubation_period + example_reporting_delay),
+  rt = rt_opts(prior = list(mean = 2, sd = 0.1), rw = 14),
+  gp = NULL,
+  obs = obs,
+  horizon = horizon,
+  output = "fit"
+)
+```
+
+For the `R` data, we will set up an arbitrary trajectory and add some Gaussian noise.
+```{r R-data}
+# Arbitrary reproduction number trajectory
+R <- c(
+  rep(2, 40), rep(0.5, 10), rep(1, 10), 1 + 0.04 * 1:20, rep(1.4, 5),
+  1.4 - 0.02 * 1:20, rep(1.4, 10), rep(0.8, 5), 0.8 + 0.02 * 1:20
+)
+# Add Gaussian noise
+R_noisy <- R * rnorm(length(R), 1, 0.05)
+```
+
+Now, we are ready to simulate the true infections and $R_t$ data by sampling from $10$ posterior samples.
+```{r true-data}
+# Forecast infections and the trajectory of Rt
+forecast <- forecast_infections(
+  estimates$estimates,
+  R = R_noisy,
+  samples = 10
+)
+```
+
+We will now extract the required data:
+- `R_true`: the median of the simulated $R_t$ values,
+- `infections_true`: the infections by date of infection, and
+- `reported_cases_true`: the reported cases by date of report.
+```{r extract-true-data}
+R_true <- forecast$summarised[variable == "R"]$median
+
+# Get the posterior samples from which to extract the simulated infections and reported cases
+posterior_sample <- forecast$samples[sample == 1]
+
+# Extract the simulated infections
+infections_true <- posterior_sample[variable == "infections"]$value
+
+# Extract the simulated reported cases and rename the "value" column to "confirm" (to match EpiNow2 requirements)
+reported_cases_true <- posterior_sample[
+  variable == "reported_cases", .(date, confirm = value)
+]
+```
+
+Now, to the main part of this vignette: we will define and run the different model options and evaluate their runtimes and estimation performance.
+
+## Model options
+
+```{r model-descriptions}
+model_descriptions <- data.table(
+  model = c(
+    "non_mechanistic",
+    "gp",
+    "gp_stationary",
+    "rw1_no_gp",
+    "rw1_gp",
+    "rw7_no_gp",
+    "rw7_gp",
+    "vb_rstan",
+    "vb_cmdstanr",
+    "pathfinder",
+    "laplace"
+  ),
+  description = c(
+    "The model with no priors on Rt",
+    "Gaussian process turned on and with a non-stationary prior on $R_t$ (default)",
+    "Gaussian process turned on and with a stationary Gaussian process prior on $R_t$",
+    "1-day random walk, Gaussian process turned off, and a non-stationary prior on $R_t$ (default)",
+    "1-day random walk, Gaussian process turned on, and a non-stationary prior on $R_t$ (default)",
+    "7-day random walk, Gaussian process turned off, and a non-stationary prior on $R_t$ (default)",
+    "7-day random walk, Gaussian process turned on, and a non-stationary prior on $R_t$ (default)",
```

Comment:
I think it might be useful to describe the models using categories vs free text (so process model changes, Rt model changes, fitting method changes, fitting package changes) if possible?

Reply:
Not sure I understand. Can you give an example by rewording one of the examples above?

Reply:
Please see the latest version in 495881519f2368faed9080eda7532640e4d20d14 and let me know what you think.